htdocs/bom/class/bom.class.php

```diff
@@ -1147,6 +1147,38 @@ public function getNetNeedsTree(&$TNetNeeds = array(), $qty = 0, $level = 0)
 			}
 		}
 	}
+
+	/**
+	 * Recursively retrieves all parent bom in the tree that leads to the $bom_id bom
+	 *
+	 * @param 	array	$TParentBom		We put all found parent bom in $TParentBom
+	 * @param 	int		$bom_id			ID of bom from which we want to get parent bom ids
+	 * @param 	int		$level		Protection against infinite loop
+	 * @return 	void
+	 */
+	public function getParentBomTreeRecursive(&$TParentBom, $bom_id = '', $level = 1)
+	{
+
+		// Protection against infinite loop
+		if ($level > 1000) {
+			return 0;
```

Comment:
Not compatible with doc saying   "return void"
Just do "return" or better, change return code in doc into   @return  int